core/src/utilities/helpers/global-search-helpers.js

```diff
@@ -1,13 +1,195 @@
 /* istanbul ignore file */
-class GlobalSearchHelperClass {
-  constructor() {}
+import { GenericHelpers } from './';
+import { KEYCODE_ARROW_UP, KEYCODE_ARROW_DOWN, KEYCODE_ENTER, KEYCODE_ESC } from './../keycode.js';
+import { Routing } from './../../services/routing';
+import { LuigiI18N } from './../../core-api';
+export class GlobalSearchHelperClass {
+  dispatch;
+  search;
+  isCustomSearchRenderer;
+  isCustomSearchResultItemRenderer;
+  luigiCustomSearchItemRenderer__slotContainer;
+
+  constructor(search, dispatcher) {
+    this.search = search;
+    this.dispatch = dispatcher;
+  }
+
+  getCustomRenderer() {
+    if (!this.search.searchProvider) return;
+    this.isCustomSearchRenderer = GenericHelpers.isFunction(this.search.searchProvider.customSearchResultRenderer);
+    this.isCustomSearchResultItemRenderer = GenericHelpers.isFunction(
+      this.search.searchProvider.customSearchResultItemRenderer
+    );
+  }
+
+  updateLuigiCustomSearchItemRenderer__slotContainer(updatedSlotContainer) {
+    this.luigiCustomSearchItemRenderer__slotContainer = updatedSlotContainer;
+  }
 
   handleVisibilityGlobalSearch() {
     if (!document.querySelector('.lui-global-search')) return;
     const globalSearchCtn = document.querySelector('.lui-global-search');
     const condition = globalSearchCtn.offsetWidth <= 384;
     globalSearchCtn.classList.toggle('lui-global-search-toggle', condition);
   }
-}
 
-export const GlobalSearchHelper = new GlobalSearchHelperClass();
+  setSearchPlaceholder(inputElement) {
+    const placeHolder = this.getSearchPlaceholder();
+    if (placeHolder) {
+      inputElement.placeholder = placeHolder;
+    }
+  }
+
+  getSearchPlaceholder() {
+    const searchProvider = this.search.searchProvider;
+    if (!searchProvider || !searchProvider.inputPlaceholder) {
+      return undefined;
+    }
+    const currentLocale = LuigiI18N.getCurrentLocale();
+    if (GenericHelpers.isFunction(searchProvider.inputPlaceholder)) {
+      return searchProvider.inputPlaceholder();
+    }
+    if (typeof searchProvider.inputPlaceholder === 'string') {
+      const translated = LuigiI18N.getTranslation(searchProvider.inputPlaceholder);
+      if (!!translated && translated.trim().length > 0) {
+        return translated;
+      }
+      return searchProvider.inputPlaceholder;
+    }
+    if (typeof searchProvider.inputPlaceholder === 'object') {
+      return searchProvider.inputPlaceholder[currentLocale];
+    }
+  }
+
+  renderCustomSearchItem(item, slotContainer, index) {
+    setTimeout(() => {
+      search.searchProvider.customSearchResultItemRenderer(item, slotContainer.children[index], searchApiObj);
+    });
+    return '';
+  }
+
+  onKeyUp({ keyCode }, displaySearchResult) {
+    if (this.search && this.search.searchProvider) {
+      if (GenericHelpers.isFunction(this.search.searchProvider.onEnter) && keyCode === KEYCODE_ENTER) {
```

Comment:
For multiple conditions `switch` statement seems to be a better choice

Reply:
Same as: https://github.com/SAP/luigi/pull/3802#discussion_r1664636537

Reply:
OK - let's leave it as is